test/component/utils/ui/circular_value_test.dart

```diff
@@ -0,0 +1,71 @@
+import "package:flutter/material.dart";
+import "package:flutter_test/flutter_test.dart";
+import "package:hooks_riverpod/hooks_riverpod.dart";
+import "package:proxima/utils/ui/circular_value.dart";
+import "package:proxima/utils/ui/error_alert.dart";
+
+void main() {
+  Widget testCircularValue(AsyncValue<void> value) => CircularValue(
+        value: value,
+        builder: (context, data) => const Text("Completed"),
+        fallbackBuilder: (context, error) => const Text("Strange Error"),
+      );
+
+  testWidgets(
+      "CircularValue should show CircularProgressIndicator when loading", (
+    tester,
+  ) async {
+    await tester.pumpWidget(
+      ProviderScope(
+        child: MaterialApp(
+          home: testCircularValue(const AsyncValue.loading()),
+        ),
+      ),
+    );
+
+    expect(find.byType(CircularProgressIndicator), findsOneWidget);
+  });
+
+  testWidgets("CicularValue should build with value when finished", (
+    tester,
+  ) async {
+    await tester.pumpWidget(
+      ProviderScope(
+        child: MaterialApp(
+          home: testCircularValue(const AsyncValue.data(null)),
+        ),
+      ),
+    );
+
```

Comment:
We might want to wait for the Widget to settle before we check it.
```suggestion
await tester.pumpAndSettle();
```

Reply:
Thank you for the suggestion, I added it.